### PR TITLE
Niagara Fox Transforms and Schema

### PR DESCRIPTION
**Pull request by zakird**

### CODE REVIEW

File: ztag/schema.py [hunk 1]
```diff
@@ -400,6 +400,28 @@
     "metadata":local_metadata
 })
 
+ztag_fox = SubRecord({
+    "version": AnalyzedString(es_include_raw=True),
+    "id": Integer(),
+    "hostname": AnalyzedString(es_include_raw=True)
+    "host_address": AnalyzedString(es_include_raw=True),
+    "app_name": AnalyzedString(es_include_raw=True),
+    "app_version": AnalyzedString(es_include_raw=True),
```
Comment (zzma) [inline ztag/schema.py]: is it a problem that some of these are not `AnalyzedString`s on the zgrab_schema side of things?

File: ztag/transforms/fox.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    subprotocol = protocols.FOX.DEVICE_ID
+
+    def _transform_object(self, obj):
+        zout = ZMapTransformOutput()
+        wrapped = Transformable(obj)
+        fox = wrapped['data']['fox']
+        if not fox['is_fox'].resolve():
+            raise errors.IgnoreObject()
+
+        instance_number = bacnet['instance_number'].resolve()
```
Comment (zzma) [inline ztag/transforms/fox.py]: s/bacnet/fox

Comment (zakird) [inline ztag/transforms/fox.py]: good call. fixed.

File: ztag/schema.py (hunk 1 above)
Comment (zakird) [inline ztag/schema.py]: It doesn't, but I'll fix. I essentially decided there are so few, let's just make all analyzed.

File: ztag/transforms/fox.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        brand_id = fox['brand_id'].resolve()
+        sys_info = fox['sys_info'].resolve()
+        auth_agent_type = fox['auth_agent_type'].resolve()
+
+        out = dict()
+		if instance_number:
+			out["instance_number"] = instance_number
+        if version:
+			out["version"] = version
+        if id_:
```
Comment (dadrian) [inline ztag/transforms/fox.py]: Indentation is off here.

File: ztag/annotations/protocols.py [hunk 4]
```diff
@@ -23,6 +23,7 @@ def __process(self, obj, meta):
     (protocols.SMTP, protocols.SMTP.STARTTLS, {"device_with_smtp":{"tags":["smtp",]}}),
     (protocols.MODBUS, protocols.MODBUS.DEVICE_ID, {"schneider_nf3000":{"tags":["modbus",]}}),
     (protocols.BACNET, protocols.BACNET.DEVICE_ID, {"device_with_bacnet":{"tags":["bacnet",]}}),
+    (protocols.FOX, protocols.FOX.DEVICE_ID, {"device_with_fox":{"tags":["fox",]}}),
```
Comment (dadrian) [inline ztag/annotations/protocols.py]: It doesn't look like device_with_fox.json exists yet.

**Comment by dadrian** [comment]
We'll need to release a new version of zsearch-definitions before we can merge this.

**Pull request merged**